### PR TITLE
docs: set up ir-docs

### PR DESCRIPTION
**Pull request by frankcalise**
## Describe your PR

- Revamps documentation to be of `ir-docs` structure for Docusaurus

## Still Needs
- [ ] Content for `docs/boilerplate-docs/Boilerplate.md`
- [ ] Content for `docs/README.md`
- [ ] Link root repo `README.md` to remote IR docs site where Ignite documentation will live

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,48 +3,52 @@
...
 
 This is the React Native boilerplate that the [Infinite Red](https://infinite.red) team uses on a day-to-day basis to build client apps. Developers who use Ignite report that it saves them two to four weeks of time on average off the beginning of their React Native project!
 
 ## Intro Video
 
 Check out the [Getting Started with Ignite](https://www.youtube.com/watch?v=KOSvDlFyg20) video for a 13 minute overview!
 
+## [Full Documentation](https://github.com/infinitered/ignite/blob/master/docs)
+
+We've put great effort into the documentation as a team, please [read through it here](https://github.com/infinitered/ignite/blob/master/docs). If you're unsure why a certain decision was made related to this boilerplate or how to proceed with a particular feature, it's likely documented. If it still isn't clear, go through the proper [help channels](#reporting-bugs--getting-help) and we always welcome PRs to improve the docs!
```
Comment (trevor-coleman) [inline README.md:22]: When `ir-docs` goes live this will be `https://docs.infinite.red/ignite

File: docs/Guide.md [hunk 2]
```diff
@@ -16,9 +20,9 @@ npx ignite-cli@latest new PizzaApp
 
 It'll walk you through several questions.
 
-Once it's up and running, you can use the Ignite CLI to [generate](./Generators.md) components, screens, MST models, and more.
+Once it's up and running, you can use the Ignite CLI to [generate](./concept-docs/Generators.md) components, screens, MST models, and more.
```
Comment (trevor-coleman) [inline docs/Guide.md:23]: is the docs postfix necessary? 

`docs/concept-docs` vs `docs/concept`

Comment (frankcalise) [reply docs/Guide.md:23]: Not really, I was just following @markrickert's Reactotron PR to keep things consistent. If you want to agree on a structure we can make sure to follow it!

Comment (markrickert) [reply docs/Guide.md:23]: I don't think `/docs/WHATEVER-docs/` is necessary. If anything, organizationally, we should prefix the folders with their position in the sidebar imho

Comment (frankcalise) [reply docs/Guide.md:23]: I like that for the most part, until you have to insert one and then have to go changing all the links. But I guess we could increment by 10 or something

Comment (markrickert) [reply docs/Guide.md:23]: https://docusaurus.io/docs/sidebar/autogenerated#using-number-prefixes

> By default, Docusaurus will remove the number prefix from the doc id, title, label, and URL paths.

However, we'd still have to update the links in the markdown files. I'll do some work on organizing things tongiht

**Comment by markrickert** [comment]
I pushed up a cleanup branch that incorporates some of the changes i'd already made locally while working on this:

- removed `-docs` postfix from folders
- moved the folder structure to the boilerplate home page
- removed "Component" from the titles of the component pages
- removed internal andhor link lists since docusauras does this for us on the right sidebar based on document headers.

**Comment by jamonholmgren** [comment]
Got the conflicts resolved. Should be mergeable once @trevor-coleman and @frankcalise are on board.

**Comment by trevor-coleman** [comment]
Took a quick look and the changes look good -- one small suggestion is to use larger gaps for the ordering numbers.

For instance instead of 1, 2, 3, you can use 100,200,300. That allows for things to be inserted later without having to edit all the files.

But I don't think it's a huge concern.

But it's not really necessary if there aren't a huge number of things in the list.

**Comment by trevor-coleman** [comment]
@frankcalise -- is this still good to go? @jamonholmgren was hoping to get this onto `ir-docs` before we launch on Monday.

**Pull request merged**